### ISSUE TITLE
Polyester option?

### ISSUE THREAD

**Issue by ChrisRackauckas** [YingboMa/RecursiveFactorization.jl#25]
Would it make sense to include a form of multithreading here?

**Comment by chriselrod** [comment]
There's also `LoopVectorization.@tturbo`, which uses `Polyester.request_threads` so that it is compatible.
Depending on the size of the Schur complements, that could help performance.

**Comment by chriselrod** [comment]
On the master branch:
```julia
julia> @benchmark schur_complement!($C0,$A,$B)
samples: 10000; evals/sample: 995; memory estimate: 0 bytes; allocs estimate: 0
ns

 (28.87 - 29.22]  ██████████████████████████████ 8113
 (29.22 - 29.56]  ██████1590
 (29.56 - 29.91]  ▏1
 (29.91 - 30.25]  ▊172
 (30.25 - 30.59]  ▍92
 (30.59 - 30.94]  ▏11
 (30.94 - 31.28]  ▏4
 (31.28 - 31.63]  ▏3
 (31.63 - 31.97]  ▏1
 (31.97 - 32.32]  ▏1
 (32.32 - 32.66]   0
 (32.66 - 33.0 ]  ▏1
 (33.0  - 33.35]   0
 (33.35 - 33.69]  ▏1
 (33.69 - 70.53]  ▏10

                  Counts

min: 28.872 ns (0.00% GC); mean: 29.172 ns (0.00% GC); median: 29.109 ns (0.00% GC); max: 70.526 ns (0.00% GC).

julia> @benchmark schur_complement_thread!($C0,$A,$B)
samples: 10000; evals/sample: 991; memory estimate: 0 bytes; allocs estimate: 0
ns

 (41.76 - 42.05]  ██████████████████████████████ 9570
 (42.05 - 42.34]   0
 (42.34 - 42.64]   0
 (42.64 - 42.93]  ▎69
 (42.93 - 43.22]  █311
 (43.22 - 43.51]  ▏18
 (43.51 - 43.81]  ▏11
 (43.81 - 44.1 ]  ▏4
 (44.1  - 44.39]  ▏3
 (44.39 - 44.68]  ▏1
 (44.68 - 44.98]  ▏1
 (44.98 - 45.27]   0
 (45.27 - 45.56]   0
 (45.56 - 45.86]  ▏2
 (45.86 - 76.92]  ▏10

                  Counts

min: 41.757 ns (0.00% GC); mean: 41.931 ns (0.00% GC); median: 41.865 ns (0.00% GC); max: 76.921 ns (0.00% GC).
```
`schur_complement!` should be unchanged, so that timing difference is noise.
The 50 -> 42 ns for the `@tturbo` should be real. I added another (earlier) check for if it isn't threading.

**Comment by YingboMa** [comment]
Closed by #28
